Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [13]:
import numpy as np
zp=np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks=np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])
np.corrcoef(zp,ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

Коэффициент корреляции =0.887. Как найти "с помощью ковариации и среднеквадратичных отклонений двух признаков" Пока не разобралась, а интернет не ловит (

In [14]:
cov=np.mean(zp*ks)-np.mean(zp)*np.mean(ks)
cov

9157.839999999997

In [15]:
cov=np.cov(zp,ks)
cov

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

In [16]:
cov=np.cov(zp,ks,ddof=0)
cov

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [6]:
import numpy as np
iq=np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
iq

array([131, 125, 115, 122, 131, 115, 107,  99, 125, 111])

In [8]:
mu=iq.mean()
mu

118.1

In [9]:
s=math.sqrt(iq.var(ddof=1))
s

10.54566788359614

In [10]:
Tt=2.262
d=Tt*s/math.sqrt(10)
d

7.543392236918348

In [ ]:
Доверительный интервал [118.1-7.543, 118.1+7.543]  или [110.557,125.643]

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [1]:
import math
M=174.2
g=math.sqrt(25)
n=27
z=1.96
d=1.96*g/math.sqrt(27)
d

1.8860108793527777

In [ ]:
Доверительный интервал [174.2-1.886, 174.2+1.886]  или [172.314,176.086]